specs/src/specs/data_structures.md

```diff
@@ -567,17 +564,17 @@ The first [`NAMESPACE_ID_BYTES`](./consensus.md#constants) of a share's raw data
 
 ### Arranging Available Data Into Shares
 
-The previous sections described how some original data, arranged into a `k * k` matrix, can be extended into a `2k * 2k` matrix and committed to with NMT roots. This section specifies how [available data](#available-data) (which includes [transactions](#transactiondata), [intermediate state roots](#intermediatestaterootdata), [evidence](#evidencedata), and [messages](#messagedata)) is arranged into the matrix in the first place.
+The previous sections described how some original data, arranged into a `k * k` matrix, can be extended into a `2k * 2k` matrix and committed to with NMT roots. This section specifies how [available data](#available-data) (which includes [transactions](#transactiondata), [intermediate state roots](#intermediatestaterootdata), PayForBlob transactions, and [messages](#messagedata)) is arranged into the matrix in the first place.
 
 Then,
 
-1. For each of `transactionData`, `intermediateStateRootData`, and `evidenceData`, [serialize](#serialization):
+1. For each of `transactionData`, `intermediateStateRootData`, PayForBlob transactions, [serialize](#serialization):
     1. For each request in the list:
         1. [Serialize](#serialization) the request (individually).
         1. Compute the length of each serialized request, [serialize the length](#serialization), and pre-pend the serialized request with its serialized length.
     1. Split up the length/request pairs into [`SHARE_SIZE`](./consensus.md#constants)`-`[`NAMESPACE_ID_BYTES`](./consensus.md#constants)`-`[`SHARE_RESERVED_BYTES`](./consensus.md#constants)-byte chunks.
     1. Create a [share](#share) out of each chunk. This data has a _reserved_ namespace ID, so the first [`NAMESPACE_ID_BYTES`](./consensus.md#constants)`+`[`SHARE_RESERVED_BYTES`](./consensus.md#constants) bytes for these shares must be [set specially](#share).
-1. Concatenate the lists of shares in the order: transactions, intermediate state roots, evidence.
+1. Concatenate the lists of shares in the order: transactions, intermediate state roots, PayForBlob transactions.
```

Comment:
On this line, I think we should differentiate transactions and PayForBlob transactions because they're written to different shares.

I see why there's some confusion: "Block data consists of transactions (which modify the Celestia chain's state)" applies to PayForBlob transactions too. Maybe we can re-word or add content to disambiguate.

Reply:
> Maybe we can re-word or add content to disambiguate.

Yes, that would be actually very helpful, otherwise, it makes this confusion that PayForBlobs are not state-changing transactions.

Reply:
resolved by https://github.com/celestiaorg/celestia-app/pull/1436/commits/0adcded858b5d531a00f7367ccef26111bdc660a